app/models/calendar_event.ts

```diff
@@ -0,0 +1,56 @@
+import vine from "@vinejs/vine";
+import { DateTime } from "luxon";
+
+import { BaseModel } from "@adonisjs/lucid/orm";
+
+import { typedColumn } from "#decorators/typed_model";
+import { preloadRelations } from "#scopes/preload_helper";
+import { handleSearchQuery } from "#scopes/search_helper";
+import { handleSortQuery } from "#scopes/sort_helper";
+
+export default class CalendarEvent extends BaseModel {
+  @typedColumn({ isPrimary: true, type: "integer" })
+  declare id: number;
+
+  @typedColumn({ type: "string", optional: true, autoGenerated: true })
+  declare googleCalId: string | null;
+
+  get isGoogleEvent(): boolean {
+    return this.googleCalId !== null;
+  }
+
+  @typedColumn.dateTime({
+    autoCreate: false,
+    autoUpdate: false,
+    autoGenerated: false,
+    validator: vine.luxonDateTime(),
+  })
+  declare startTime: DateTime;
+
+  @typedColumn.dateTime({
+    autoCreate: false,
+    autoUpdate: false,
+    autoGenerated: false,
+    validator: vine.luxonDateTime(),
+  })
+  declare endTime: DateTime;
```

Comment:
`vine.luxonDateTime()` is the new default validator for `@typedColumn.datetime` columns - passing it here is redundant (as are all other properties passed to the decorator tbh - those should all be defaults)